client/files.go

```diff
@@ -110,3 +115,56 @@ func WriteUserSignedKey(signedKey string, key string) (err error) {
 	_, err = signedKeyFile.WriteString(signedKey)
 	return err
 }
+
+// FindUserPubKeys checks every pubkey in `keys` list and returns only existsing keys (or error if
+// all pubkeys don't exist)
+func FindUserPubKeys(keys []string) ([]string, error) {
+	var found = []string{}
+	for _, key := range keys {
+		pubKeyPath, err := UserPubKeyExists(key)
+		if err != nil {
+			return nil, err
+		}
+		if pubKeyPath != "" {
+			found = append(found, pubKeyPath)
+		}
+	}
+
+	if len(found) == 0 {
+		var errStr = fmt.Sprintf(`user SSH keys at %s doesn't exist.
+
+Please generate at least one with command like this :
+
+`, strings.Join(keys, ", "))
+
+		// keys list is not explicitly set, suggest generating ed25519 as default
+		if reflect.DeepEqual(keys, DefaultSSHKeys) {
+			errStr += "\tssh-keygen -f ~/.ssh/id_ed25519 -t ed25519\n"
+		} else {
+			for _, key := range keys {
+				suggestedSSHKey := strings.Replace(key, ".pub", "", 1)
+				errStr += fmt.Sprintf("\tssh-keygen -f %s -t %s\n", suggestedSSHKey, chooseSSHKeyType(suggestedSSHKey))
+			}
+		}
+
+		return nil, fmt.Errorf(errStr)
+	}
+	return found, nil
+}
+
+func chooseSSHKeyType(key string) string {
+	switch {
+	case strings.Contains(key, "dsa"):
+		return "dsa"
+	case strings.Contains(key, "ecdsa_sk"):
+		return "ecdsa-sk"
+	case strings.Contains(key, "ecdsa"):
+		return "ecdsa"
+	case strings.Contains(key, "ed25519_sk"):
+		return "ecdsa-sk"
+	case strings.Contains(key, "rsa"):
+		return "rsa"
```

Comment:
Same as for dsa, a deprecation message could be helpful for the user

Reply:
I'm not sure that we should mark rsa type as completly deprecated. Actually, only sha1 type is deprecated, but rsa-sha2-256 and rsa-sha2-512 are still valid. And after `x/crypto/ssh` dependency is fixed, rsa-sha2-512-cert-v01@openssh.com and rsa-sha2-256-cert-v01@openssh.com certificate types should also be valid.

Changed type to `rsa-sha2-512`, I think it is enough.

Reply:
Yes, you're right we shouldn't mark `rsa` as deprecated.

I dig a bit into "-t" parameter because it was not clear to me what it is exactly doing. I found a detailed answer [here](https://superuser.com/questions/1556852/how-to-check-if-your-ssh-keys-are-in-the-ssh-rsa2-format/1556861#1556861).

>> the default option rsa for the -t argument explains that the chosen option is using the SHA1 signature, so one should choose rsa-sha2-256 for example.
>
>This applies when issuing certificates, but is irrelevant when generating plain keys.

What I understand is that for generating keys, like in the example we display, there is no signature in the process. So that the argument for "-t" should still be "rsa".

The man page of "ssh-keygen" seems to say the same thing:

>
      -t dsa | ecdsa | ecdsa-sk | ed25519 | ed25519-sk | rsa
             Specifies the type of key to create.  The possible values
             are “dsa”, “ecdsa”, “ecdsa-sk”, “ed25519”, “ed25519-sk”, or
             “rsa”.

             This flag may also be used to specify the desired signature
             type when signing certificates using an RSA CA key.  The
             available RSA signature variants are “ssh-rsa” (SHA1
             signatures, not recommended), “rsa-sha2-256”, and
             “rsa-sha2-512” (the default).

Reply:
I tested this case:

```bash
$ ssh-keygen -f ./id_rsa -t rsa-sha2-512
        Type: ssh-rsa-cert-v01@openssh.com user certificate
        Signing CA: RSA SHA256:... (using rsa-sha2-512)
```

So it seems to me that it works fine in case of ssh client, and we can keep this type.

> This applies when issuing certificates, but is irrelevant when generating plain keys.

signmykey doesn't use ssh-keygen to issue certificates, but I tested it too, and results rather interesting:

```
# -t rsa doesn't work
$ ssh-keygen -s ./local-ca/user-ca -I test -t rsa ./ssh/id_rsa.pub 
Couldn't certify key ./ssh/id_rsa.pub: invalid argument

# without -t result the same - "using rsa-sha2-512"
$ ssh-keygen -s ./local-ca/user-ca -I test -t rsa-sha2-512 ./ssh/id_rsa.pub
        Type: ssh-rsa-cert-v01@openssh.com user certificate
        Signing CA: RSA SHA256:... (using rsa-sha2-512)

$ ssh-keygen -s ./local-ca/user-ca -I test -t rsa-sha2-256 ./ssh/id_rsa.pub
        Type: ssh-rsa-cert-v01@openssh.com user certificate
        Signing CA: RSA SHA256:... (using rsa-sha2-256)

$ ssh-keygen -s ./local-ca/user-ca -I test -t ssh-rsa ./ssh/id_rsa.pub
        Type: ssh-rsa-cert-v01@openssh.com user certificate
        Signing CA: RSA SHA256:... (using ssh-rsa)
```

Reply:
> So that the argument for "-t" should still be "rsa".

I can change type to "rsa" if you want, it's not a problem

Reply:
Yes sorry, it was a long comment to just say that rolling back to "rsa" as in your first commit is better, thanks :)